flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliClient.java

```diff
@@ -125,21 +137,37 @@ void executeInInteractiveMode(LineReader lineReader) {
     }
 
     /** Opens the non-interactive CLI shell. */
-    public void executeInNonInteractiveMode(String content) {
+    public void executeInNonInteractiveMode(URI uri) {
         try {
             terminal = terminalFactory.get();
-            executeFile(content, terminal.output(), ExecutionMode.NON_INTERACTIVE_EXECUTION);
+            if (isApplicationMode(executor.getSessionConfig())) {
+                String scheme = StringUtils.lowerCase(uri.getScheme());
+                String clusterId;
+                // local files
+                if (scheme == null || scheme.equals("file")) {
```

Comment:
Would it be better if we could support shipping local files to the remote? That said, I think the current implementation is also fine.

Reply:
Here we just read the sql script file if the file is in local fs. It's not an easy work to ship files in the k8s deployment. It requires users to configure a remote oss env.